pennylane/hf/integrals.py

```diff
@@ -461,3 +462,392 @@ def kinetic_integral(*args):
+    Hermite Coulomb integral, respectively. The sum goes over :math:`i + j + 1`, :math:`k + l + 1`
+    and :math:`m + m + 1` for :math:`t`, :math:`u` and :math:`v`, respectively, and :math:`p` is
+    computed from the exponents of the two Gaussian functions as :math:`p = \alpha + \beta`.
+
+    Args:
+        la (tuple[int]): angular momentum for the first Gaussian function
+        lb (tuple[int]): angular momentum for the second Gaussian function
+        ra (array[float]): position vector of the the first Gaussian function
+        rb (array[float]): position vector of the the second Gaussian function
+        alpha (array[float]): exponent of the first Gaussian function
+        beta (array[float]): exponent of the second Gaussian function
+        r (array[float]): position vector of nucleus
+
+    Returns:
+        array[float]: nuclear attraction integral between two Gaussian functions
+    """
+    l1, m1, n1 = la
+    l2, m2, n2 = lb
+    p = alpha + beta
+    rgp = (alpha * ra[:, anp.newaxis, anp.newaxis] + beta * rb[:, anp.newaxis, anp.newaxis]) / (
+        alpha + beta
+    )
+    dr = rgp - anp.array(r)[:, anp.newaxis, anp.newaxis]
+
+    a = 0.0
+    for t in range(l1 + l2 + 1):
+        for u in range(m1 + m2 + 1):
+            for v in range(n1 + n2 + 1):
+                a = a + expansion(l1, l2, ra[0], rb[0], alpha, beta, t) * expansion(
+                    m1, m2, ra[1], rb[1], alpha, beta, u
+                ) * expansion(n1, n2, ra[2], rb[2], alpha, beta, v) * _hermite_coulomb(
+                    t, u, v, 0, p, dr
+                )
+    a = a * 2 * anp.pi / p
+    return a
+
+
+def generate_attraction(r, basis_a, basis_b):
+    r"""Return a function that computes the nuclear attraction integral for two contracted Gaussian
+    functions.
+
+    Args:
+        r (array[float]): position vector of nucleus
+        basis_a (BasisFunction): first basis function
+        basis_b (BasisFunction): second basis function
+
+    Returns:
+        function: function that computes the nuclear attraction integral
+
+    **Example**
+
+    >>> symbols  = ['H', 'H']
+    >>> geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]], requires_grad = False)
+    >>> alpha = np.array([[3.425250914, 0.6239137298, 0.168855404],
+    >>>                   [3.425250914, 0.6239137298, 0.168855404]], requires_grad = True)
+    >>> mol = Molecule(symbols, geometry, alpha=alpha)
+    >>> basis_a = mol.basis_set[0]
+    >>> basis_b = mol.basis_set[1]
+    >>> args = [mol.alpha]
+    >>> generate_attraction(geometry[0], mol.basis_set[0], mol.basis_set[1])(*args)
+    0.801208332328965
+    """
+
+    def attraction_integral(*args):
+        r"""Compute the nuclear attraction integral for two contracted Gaussian functions.
+
+        Args:
+            args (array[float]): initial values of the differentiable parameters
+
+        Returns:
+            array[float]: the nuclear attraction integral between two contracted Gaussian orbitals
```

Comment:
I keep adding the `electron-nuclear` adjective here for the sake of clarity. However if the term "nuclear attraction" is the right one, feel free to ignore this suggestions. But I find counter intuitive "nuclear attraction"
```suggestion
            array[float]: the electron-nuclear attraction integral between two contracted Gaussian orbitals
```